### PR TITLE
Metadata Walkthrough

### PR DESCRIPTION
**Pull request by peternied**
### Description
Please focus first on the readme which is what would be demo'ed:
- `MetadataMigration/README.md` [[link]](https://github.com/opensearch-project/opensearch-migrations/pull/812/files#diff-0e7ba33e2f4c872a6ea410a51af9a377498d255e4d0353a1cdbf330f52e249ca)

Then please look at the test cases which are directly from the demo script
- `MetadataMigration/src/test/java/org/opensearch/migrations/metadata/DemoTests.java` [[link]](https://github.com/opensearch-project/opensearch-migrations/pull/812/files#diff-ff346d1f7a6efeb64a917ffa6c740da0677a9c52b510790408f737b58c34c421)

All other changes need to be properly built out and aren't worth digging into individually.

### Issues Resolved
- Resolves https://opensearch.atlassian.net/browse/MIGRATIONS-1780

### Testing
New test case included.

### Check List
- [X] New functionality includes testing
  - [ ] ~All tests pass, including unit test, integration test and doctest~
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: MetadataMigration/README.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+### Configure
+By inspecting the source and the items it contains some options can be automatically configured.  As some items require manually intervention to what metadata items should be migrated or how to resolve different kinds of conflicts updating the configuration narrows down this scope.
+
+### Migrate
+Following the configuration used in the evaluation the target cluster is updated with all of the metadata items. During the deployment there might have been issues these can be better understood by inspecting the output and/or detailed logs.
+
+### Verification
+Before starting data migration the target cluster can be inspected by running queries to confirm the state or any other checks before the longer process of migrating data from the source to the target cluster.
+
+## Walkthrough
```
Comment (peternied) [inline MetadataMigration/README.md:37]: How is the tool installed? Need to figure out how to distribute / consume.

Possible options:
- (Default) Ship jar, include details on how to get a copy from opensearch.org website.
- Docker image, we already build docker images, we could publish the metadata image [1]

---
- [1] Turns out that OpenSearch-Benchmark already does this  https://opensearch.org/docs/latest/benchmark/user-guide/installing-benchmark/

Comment (mikaylathompson) [reply MetadataMigration/README.md:37]: I like the idea to do a docker image. We could include a (possibly stripped down?) version of the console library too, and then we don't really have to worry about two versions of an interface.

Comment (peternied) [reply MetadataMigration/README.md:37]: I've created a Jira task to track this https://opensearch.atlassian.net/browse/MIGRATIONS-1851

**Review by mikaylathompson — APPROVED** [review]
LGTM

**Pull request merged**